Migration-V2.md

```diff
@@ -0,0 +1,67 @@
+
+```
+"github.com/IBM/go-sdk-core/v4/core"
+```
+
+### Breaking Version Date Type Change
+
+For all packages, the `Version` property on the service struct is now a `*String`. It had previously been a `String` which was provided as a literal. The Go SDK Core has a convenience method to provide this string pointer:
+
+```go
+service, err = assistantv1.NewAssistantV1(&assistantv1.AssistantV1Options{
+	Version:     core.StringPtr("2020-04-01"),
+	ServiceName: "assistant",
+ })
+```
+
+### Breaking Changes to model setter functions
+
+In the `1.x` branch of the SDK, setter methods were provided for almost all model structs for dealing with data from the service. Version `2.0.0` makes significant breaking changesin how setters are handled for these structs.
+
+Specifically, now the only structs that have setters for their properties are structs with the naming convention `{ServiceOperation}Options`. These are called `Options Models` and are typically the top level struct that is passed into a function. These structs have setter methods exposed that allow the individual properties within them to be set.
+
+For lower level models underneath these options models, they had setter methods for each individual property. As this bloated the code surface and provided little value compared to setting the property directly, we have decided in version `2.0.0` to remove these setters.
+
+If your code editor calls out that a setter method you are using has been removed, the appropriate way to update is to set the property directly.
+
+### Providing custom `Context` to functions
+
+Version `2.0.0` of the Go SDK now adds alternate versions to all service functions, named `{function}WithContext`. This function calls the expected service function, but accepts a `Context` as the first parameter. This allows you to provide a context that you would like the function to execute in.
+
+Internally, the Go SDK uses these functions for all operations, but the regular function provides the background context as a default.
+
+### Enabling `gzip` compression
+
+All service structs now expose a function `GetEnableGzipCompression()` and `SetEnableGzipCompression(enableGzip bool)`. These methods allow you to enable `gzip` compression within the service methods.
+
+### Automatic Retries
+
+All service structs will now automatically retry HTTP methods in the event of an error.
+
+You can customize the retry logic to suit your purposes. The function `EnableRetries(maxRetries int, mayTretryInterval time.Duration)` allows you to configure the retry strategy that you would like to enable for the functions called by the service.
+
+Optionally, you can disable retries by using the `DisableRetries()` method exposed on the service struct.
+
+## Service specific changes
+
+### assistantv1
+
+- In the `Message()` function, the MessageInput{}'s `Text` property is now a `*string`. This can be constructed via the `core.StringPtr("{message}")` convenience method.
```

Comment:
FYI... the core.XXXPtr() methods are typically used for taking the address of a literal (e.g. `core.StringPtr("string literal")`) but if you have a variable in your app, then just take the address of it:
```
     var myMessage = "message text"
     newModel := &MessageInput{MessageText: &myMessage}
```